src/modules/frequency/application/frequency.service.ts

```diff
@@ -1,26 +1,258 @@
-import { Inject, Injectable } from "@nestjs/common";
-import { FREQUENCY_QUERIES_TOKEN, IFrequencyQueries } from "./frequency.service.query.interfaces";
-import { ResponseGetMyClassesDTO } from "./frequency.dtos";
+import {
+  BadRequestException,
+  ConflictException,
+  Inject,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from "@nestjs/common";
+import {
+  FREQUENCY_QUERIES_TOKEN,
+  IFrequencyQueries,
+} from "./frequency.service.query.interfaces";
+import {
+  UserClassesResponseDTO,
+  UpdateGeneralAttendanceRequestDTO,
+  UpdateGeneralAttendanceItemDTO,
+  StudentListByClassAndDateResponseDTO,
+  UpdateClassAttendanceRequestDTO,
+  GeneralAttendanceResponseDTO,
+} from "./dtos";
+import { Frequency } from "../domain/frequency.entity";
+import {
+  FREQUENCY_REPOSITORY_TOKEN,
+  IFrequencyRepository,
+} from "../domain/frequency.repository";
+import { FrequencyStatus } from "src/common/enums/domain.enums";
 
 @Injectable()
-export class FrequencyService{
-    @Inject(FREQUENCY_QUERIES_TOKEN)
-    private readonly frequencyQueryService: IFrequencyQueries;
-    public async getUserClasses(userId: number): Promise<ResponseGetMyClassesDTO>{
-        var userClasses = await this.frequencyQueryService.getMyClasses(userId);
-        userClasses.push({
-            classId: null,
-            className: "Geral",
-            classState: "ATIVA",
-            levelName: null,
-            isGeral: true,
-            activity: {
-                activityId: null,
-                activityName: "Atividade geral",
-            },
-        })
-        return {
-            classes : userClasses
-        };
-    }
-}
\ No newline at end of file
+export class FrequencyService {
+  @Inject(FREQUENCY_QUERIES_TOKEN)
+  private readonly frequencyQueryService: IFrequencyQueries;
+  @Inject(FREQUENCY_REPOSITORY_TOKEN)
+  private readonly frequencyRepository: IFrequencyRepository;
+
+  public async getUserClasses(userId: number): Promise<UserClassesResponseDTO> {
+    const userClasses = await this.frequencyQueryService.getMyClasses(userId);
+    userClasses.push({
+      classId: null,
+      className: "Geral",
+      classState: "ATIVA",
+      levelName: null,
+      isGeral: true,
+      activity: {
+        activityId: null,
+        activityName: "Atividade geral",
+      },
+    });
+    return {
+      classes: userClasses,
+    };
+  }
+  public async getGeneralAttendance(
+    date: Date,
+  ): Promise<GeneralAttendanceResponseDTO> {
+    const studentList =
+      await this.frequencyQueryService.getGeneralAttendance(date);
+    return {
+      date: date.toISOString().split("T")[0],
+      studentList: studentList,
+    };
+  }
+  public async updateGeneralAttendance(
+    object: UpdateGeneralAttendanceRequestDTO,
```

Comment:
Since `Object` is a reserved keyword, maybe consider renaming it to avoid confusion.
Just a suggestion
```suggestion
    data: UpdateGeneralAttendanceRequestDTO,
```

Reply:
De acordo.